internal/routes/all_routes.go

```diff
@@ -64,6 +64,9 @@ func apiRouter() http.Handler {
 			r.Route("/{type}", func(r chi.Router) {
 				r.Post("/", s.CreateReservation)
 			})
+			r.Route("/{ID}/status", func(r chi.Router) {
```

Comment:
I had to add the `status` suffix due to `reservations/{ID}` collides with `reservations{type}`

Reply:
I don't mind the status suffix personally :)

Reply:
This smells like a dirty trick, let's should push for a clean REST design by refactoring the existing endpoint which is in our way.

We briefly discussed use of URLs or URL params for filtering and I think this case shows that using URL params for filtering is a good approach. I would suggest to refactor `/{type}` to be `/?type=XXX` URL param so we can use the RESTy `/{ID}` for status.

Reply:
@amirfefer I also belive the putting this route first would fix the issue if we allow get only there, as I'm suggesting below, it should not match the POST requests ;) which won't be more work and will be tidier :)

Reply:
Please remove the '/status' and simply use 'GET /{ID}'